webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/ViewTabContent.elm

```diff
@@ -115,10 +121,10 @@ tabContent model details =
                     [ input[ id "rule-tags-key", type_ "text", placeholder "key", class "form-control", onInput (\s -> UpdateRuleForm {details | ui = {ui | newTag = {newTag | key = s}}} ), value newTag.key][]
                     , span [ class "input-group-addon addon-json"][ text "=" ]
                     , input[ type_ "text", placeholder "value", class "form-control", onInput (\s -> UpdateRuleForm {details | ui = {ui | newTag = {newTag | value = s}}}), value newTag.value][]
-                    , span [ class "input-group-btn"][ button [ class "btn btn-success", type_ "button", onClick  (UpdateRuleForm {details | rule = {rule |  tags = newTag :: rule.tags }}) ][ span[class "fa fa-plus"][]] ]
+                    , span [ class "input-group-btn"][ button [ class "btn btn-success", type_ "button", onClick  (UpdateRuleForm {details | rule = {rule |  tags = newTag :: rule.tags }, ui = {ui | newTag = {newTag | key = "", value = "" }}}), disabled (String.isEmpty details.ui.newTag.key || String.isEmpty details.ui.newTag.value) ][ span[class "fa fa-plus"][]] ]
```

Comment:
you don't need to copy newTag here: 
```ui = {ui | newTag = {newTag | key = "", value = "" }} ```

you could 

```ui = {ui | newTag = Tag "" ""}```

(or the correct type, but don't need to make it so complicated)